inspirehep/modules/crossref/core.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""CrossRef Core."""
+
+from __future__ import absolute_import, division, print_function
+
+from flask import current_app, request, jsonify
+
+import requests
+from urlparse import urljoin
```

Comment:
Nit: it's not enforced by anything, but the imports are separated in sections according to some rules. In general:

1. Future imports
2. Standard library imports
3. Vendored from outside CERN
4. Vendored from inside CERN
5. INSPIRE packages (external)
6. INSPIRE packages (internal)

Moreover, imports are sorted alphabetically, but imports of whole modules come first. All in all, here it becomes:
```python
from __future__ import absolute_import, division, print_function

from urlparse import urljoin

import requests
from flask import current_app, jsonify, request
```

Reply:
Although it's better for Python 3 compatibility to use `six` here, since `urlparse` was moved. So in fact:
```python
from __future__ import absolute_import, division, print_function

import requests
from flask import current_app, jsonify, request
from six.moves.urllib.parse import urljoin
```